Assets/Plugins/Source/Editor/Windows/LogLevelWindow.cs

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2024 PlayEveryWare
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+using UnityEngine;
+using UnityEditor;
+using System;
+using System.Collections.Generic;
+
+#if !EOS_DISABLE
+namespace PlayEveryWare.EpicOnlineServices.Editor.Windows
+{
+    public class LogLevelWindow : EOSEditorWindow
+    {
+        LogLevelConfig currentLogLevelConfig;
+
+        string[] categories = LogLevelHelper.LogCategoryStringArray;
+        string[] levels = LogLevelHelper.LogLevelStringArray;
+
+        int selectedCategoryIndex = 0;
+        bool showAllCategories = false;
+        string previousAllCategoryLogLevel;
+
+        [MenuItem("Tools/EOS Plugin/Log Level")]
+        public static void ShowWindow()
+        {
+            GetWindow<LogLevelWindow>("Log Level");
+        }
+        protected override void Setup()
+        {
+            currentLogLevelConfig = EpicOnlineServices.Config.Get<LogLevelConfig>();
+            if (currentLogLevelConfig.logCategoryLevelPairs == null) 
+            {
+                currentLogLevelConfig.logCategoryLevelPairs = new List<LogCategoryLevelPair>();
+            }
+
+            if (currentLogLevelConfig.logCategoryLevelPairs.Count != categories.Length)
+            {
+                currentLogLevelConfig.logCategoryLevelPairs.Clear();
+
+                foreach (var category in categories)
+                {
+                    currentLogLevelConfig.logCategoryLevelPairs.Add(new LogCategoryLevelPair(category, "Info"));
+                }
+
+                SaveToJSONConfig(true);
```

Comment:
Why are you saving the JSON inside the `Setup` function?

Reply:
This is under the if statement when the list of categories has changed (for scenarios like a new interface from SDK),
and it regenerates a new config file in such cases.

Reply:
I see - so basically - when the Window is opened, it loads the current log level configuration - then checks to see if there are any log levels that aren't included in it - and if there are, it updates them accordingly and saves the JSON. 

Am I understanding that correctly?

Reply:
Partially. It creates a new config and set them all to LogLevel.Info.
The usage of this is pretty rare, since EOSSDK wouldn't dish out new categories that often.